### DISCUSSION TITLE
Configuring Postgres Service Container

### DISCUSSION THREAD

**Discussion by ml-sherpa** [community/community#106564] [Actions] [answered]
### Select Topic Area

Question

### Body

What's the best way to configure a postgres service container in github actions. Specifically I want to configure [max_locks_per_transaction](https://www.postgresql.org/docs/14/runtime-config-locks.html#max-locks-per-transaction) when the container starts. 

- I can't use an [environment variable](https://www.postgresql.org/docs/14/libpq-envars.html) as these are not used for configuring the server.
- I can't mount a volume because services are started before the repo is checked out so I can't mount a `postgresql.conf` file. 
- The `docker container create` doesn't seem to have a way to [pass command arguments](https://docs.docker.com/engine/reference/commandline/container_create/) so I can't specify `CMD ["postgres", "-c", "max_locks_per_transaction=1024"]`
- `docker container create` will accept `--entrypoint` but I have the same issue as I had trying to mount `postgresql.conf`. The repo is not checked out.

**Comment by ghost** [comment, marked as answer]
Hi. Since environment variables and mounting volumes are not feasible options for you, one workaround could be to create a custom Docker image with the desired PostgreSQL configuration and then use that image in your GitHub Actions workflow. You can pre-configure the `max_locks_per_transaction` setting in the Dockerfile itself.

1. Now, please create a Dockerfile in your project directory:

```Dockerfile
FROM postgres:latest

# Set max_locks_per_transaction
RUN echo "max_locks_per_transaction = 1024" >> /etc/postgresql/postgresql.conf
```

2. Build the Docker image:

```bash
docker build -t my-custom-postgres .
```

3. Push the Docker image to a container registry accessible to your GitHub Actions workflow.

4. In your GitHub Actions workflow, use the custom Docker image instead of the default PostgreSQL image:

```yaml
jobs:
  build:
    runs-on: ubuntu-latest

    services:
      postgres:
        image: my-custom-postgres
        ports:
          - 5432:5432
        options: --name postgresql

    steps:
      # Your steps here
```

By using a custom Docker image with the desired PostgreSQL configuration, you can ensure that the `max_locks_per_transaction` setting is applied when the container starts, even in the context of GitHub Actions.

**Reply by ml-sherpa** [reply]
Ya, that's the route I went down

```Dockerfile
FROM postgres:latest
CMD ["postgres", "-c", "max_locks_per_transaction=1024"]
```

**Reply by jshields** [reply]
Note when going down this path that due to other missing functionality in GitHub Actions, it is not possible to securely use OIDC with a privately hosted custom image (such as on ECR). The temporary credentials will be unmasked as plaintext and potentially exposed in logs.
https://github.com/orgs/community/discussions/25225#discussioncomment-9172039

**Reply by umtdemr** [reply]
>     services:
>       postgres:
>         image: my-custom-postgres
>         ports:
>           - 5432:5432
>         options: --name postgresql

this name option saved my day. thank you
